pkg/docker/docker.go

```diff
@@ -439,7 +538,7 @@ func (d *stiDocker) CheckAndPullImage(name string) (*docker.Image, error) {
 	}
 
 	image, err := d.CheckImage(name)
-	if err != nil && err.(errors.Error).Details != docker.ErrNoSuchImage {
+	if err != nil && dockerapi.IsErrImageNotFound(err) {
```

Comment:
On Thu, Aug 4, 2016 at 5:41 PM, Ben Parees <notifications@github.com> wrote:

> In pkg/docker/docker.go
> <https://github.com/openshift/source-to-image/pull/554#discussion_r73608220>
> :
>
> > @@ -439,7 +538,7 @@ func (d *stiDocker) CheckAndPullImage(name string) (*docker.Image, error) {
> >  	}
> >
> >  	image, err := d.CheckImage(name)
> > -	if err != nil && err.(errors.Error).Details != docker.ErrNoSuchImage {
> > +	if err != nil && dockerapi.IsErrImageNotFound(err) {
>
> so you're calling d.CheckImage which in turn is calling kube.InspectImage
> and kube.InspectImage is eating the dockerapi imageNotFoundError and
> instead returning a kube.imageNotFoundError.
>
> it's the kube.imageNotFoundError that has "no such image" (lowercase) as
> its message output.
>
Yep, my earlier scan for "no such image" did not include k8s.

> so as expected, you're getting a kube.imageNotFoundError.
>
> but you're right, i thought i saw a method for checking the error type in
> kube, but i was wrong, it was something else, i don't see any way to
> confirm the error type other than, as you said, checking the message text,
> which is bad.
>
agreed (bad) but no choice under current circumstances

> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/source-to-image/pull/554/files/63881032417630cb32a9bf66a2b033fee54d2aff..068c1c2bd2a589952a1bd740f976bca1c1fd8657#r73608220>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADbadGbXvcSHAuyIn8HPc9-V0F95Rv5Mks5qclykgaJpZM4JUVnn>
> .
>